cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/nosql/NoSqlStructuredTable.java

```diff
@@ -126,6 +129,29 @@ public CloseableIterator<StructuredRow> scan(Range keyRange, int limit) throws I
                              limit);
   }
 
+  @Override
+  public CloseableIterator<StructuredRow> scan(Range keyRange, int limit, Field<?> orderByField, SortOrder sortOrder)
+    throws InvalidFieldException, IOException {
+    LOG.trace("Table {}: Scan range {} with limit {}, sort field {} and sort order {}", schema.getTableId(),
+              keyRange, limit, orderByField, sortOrder);
+    SortedIterator sortedIterator = new SortedIterator(Collections.singleton
+      (new ScannerIterator(getScanner(keyRange), schema)).iterator(), orderByField, sortOrder, schema);
+    return new LimitIterator(Collections.singleton(sortedIterator).iterator(), limit);
+  }
+
+  @Override
+  public CloseableIterator<StructuredRow> scan(Range keyRange, int limit, Field<?> filterIndex)
+    throws InvalidFieldException, IOException {
+    LOG.trace("Table {}: Scan range {} with limit {} and index {}", schema.getTableId(), keyRange, limit, filterIndex);
+    fieldValidator.validateField(filterIndex);
+    if (!schema.isIndexColumn(filterIndex.getName())) {
+      throw new InvalidFieldException(schema.getTableId(), filterIndex.getName(), "is not an indexed column");
+    }
+    FilterByIndexIterator filterByIndexIterator = new FilterByIndexIterator(Collections.singleton
```

Comment:
nit: this is a bit hard to read with all the parentheses, can you break it up into 2 statements?